### DISCUSSION TITLE
[Feature Request]: Add a new command to print without the color theme.

### DISCUSSION THREAD

**Discussion by stevenlis** [PDConSec/vsc-print#277] [Ideas]
Hi, @PeterWone. I apologize for bringing this up again.

I have to print numerous .log files. However, the color coding in these files is unpredictable because different programs generate them differently. I am currently using the solution mentioned in https://github.com/PDConSec/vsc-print/issues/167#issuecomment-1296035679. Unfortunately, it is quite cumbersome to perform this task every time. Additionally, if I select "plain text", the line numbers will be omitted in the PDFs.

I was wondering if you could please consider adding a command to print the files without the color theme. This way, users would have the flexibility to choose how they want to print their files on the fly.

**Comment by PeterWone** [comment]
Hi Steven. I've moved your issue because there is a place for feature requests, so I don't completely forget about them.

First a look at how Print works. 

Ignoring how print sessions are managed, at some point a print session decides to render a file (or editor buffer). There are various document renderers, the default being the source code renderer. This is the one that does syntax colouring and line numbers. There's another one for Markdown and one for plaintext (no line numbers or colouring). Renderers are selected according to the content language as determined by vscode. Usually, the problem is improper language identification but, in your case, there is a definite language identifier `log`, with the complication that this covers a multitude of sins. Highlightjs is a third-party library that parses and renders as HTML with syntactic CSS classes to which I apply CSS to produce syntax colouring. This is not giving you consistent results.

I probably _can_ inhibit syntax colouring. If I don't process the content with highlightjs I'll have to HTML encode it. Currently this is a side effect of the syntax colouring step, but I could do it directly.

This is a straightforward change, but it affects a lot of code. What can you do in the meantime? 

You could fork the repo and create your own custom build with a "No syntax colouring" stylesheet that doesn't define any of the syntax classes. If I were to do that, I would get a bad review within ten minutes of release from someone who thinks having to change stylesheets is the worst thing ever.

Adding a way to assign stylesheets to languages would solve this and it's not a bad idea but it's yet another backlog item. And the people who do the most complaining about small things are also the last people to RTFM, so they'd never learn about that feature. But enough grumbling about minority users. 

Adding a way to suppress syntax colouring by not using highlightjs really needs to be per-language too, so maybe it would be better to support the null stylesheet and the language association. That addresses no-colouring as a special case of stylesheet per language. What are your thoughts on these options?

**Reply by stevenlis** [reply]
Hi, Peter. Thanks a lot for the detailed explanations. Honestly, I'm not very familiar with the technology behind how this "print" command works. But ideally, one could simply use "print without color" when wanting a printout without syntax highlighting, and use "print" if highlighting is desired, for any file types. Maybe we can add a setting like the following to allow users to define their desired color for each file type?

```
"print.colourScheme": {
    "*": "Atom One",         // global setting
    "*.py": "Ascetic",       // only apply to python files
    "*.log": null            // only apply to .log files
}
```

If this is feasible, it would be a great improvement. ~~If suppressing syntax coloring is too complicated, perhaps users could simply use "Ascetic" for files where they don't want any color.~~
